scripts/beck/st/error_injection.py

```diff
@@ -164,128 +176,156 @@ def ReadMetaData(offset):
 	with open(filename, "rb") as mdata:
 		mdata.seek(offset)
 		data=binascii.hexlify((mdata.read(8))[::-1])
-		print("** Metadata at offset {} is {}".format(hex(offset), data))
+		logger.info("** Metadata at offset {} is {}".format(hex(offset), data))
 
 def induceCorruption(recordType, noOfCorr):
-	print(recordType)
-	print("No of Error want to induce: {}".format(noOfCorr))
+	count = 0
+	read_metadata_file()
+	logger.info(recordType)
+	logger.info("No of Error want to induce: {}".format(noOfCorr))
 	lookupList = recordDict[recordType]
 	if (len(lookupList) & noOfCorr) == 0:
-		print("Record List is empty. Please choose another Record")
+		logger.error("Record List is empty. Please choose another Record")
+		count = 0
+		return count
 	elif len(lookupList) < noOfCorr:
-		print(" Record List contains Less no of entry than Input. Please reduce the number of Error Injection")
+		logger.error(" Record List contains Less no of entry than Input. Please reduce the number of Error Injection")
+		count = 0
+		return count
 	else:
-		print(lookupList)
-		print("**** Inducing {} Error in Record: {} ****".format(noOfCorr,recordType))
+		logger.info(lookupList)
+		logger.info("**** Inducing {} Error in Record: {} ****".format(noOfCorr,recordType))
 		for i in range(noOfCorr):
-			offset = lookupList[i]
+			offset = lookupList[i]  ## Please add offset here for starting from middle of offset list
 			ReadMetaData(offset+8)
-			EditMetadata(offset+8)			
+			EditMetadata(offset+8)
```

Comment:
I can not comment on existing code of "EditMetadata()".
But I prefer to remove the calling of "ReadMetaData" at the end of "EditMetadata()"
Otherwise this is very confusing.

Then we can do:
```
ReadMetaData(offset+8)   # Read Oringal
EditMetadata(offset+8)     # Modify
ReadMetaData(offset+8)   # Verify
```

Reply:
Thats correct 👍 I will add a comment in my script